server/services/courseService.test.ts

```diff
@@ -133,6 +135,54 @@ describe('CourseService', () => {
     })
   })
 
+  describe('getParticipation', () => {
+    it('returns a given course participation', async () => {
+      const courseParticipation = courseParticipationFactory.build()
+
+      when(courseClient.findParticipation).calledWith(courseParticipation.id).mockResolvedValue(courseParticipation)
+
+      const result = await service.getParticipation(token, courseParticipation.id)
+
+      expect(result).toEqual(courseParticipation)
+
+      expect(courseClientBuilder).toHaveBeenCalledWith(token)
+      expect(courseClient.findParticipation).toHaveBeenCalledWith(courseParticipation.id)
+    })
+
+    describe('when the course client throws a 404 error', () => {
+      it('returns `null`', async () => {
+        const clientError = createError(404)
+        courseClient.findParticipation.mockRejectedValue(clientError)
+
+        const notFoundCourseParticipationId = 'NOT-FOUND'
+
+        const result = await service.getParticipation(token, notFoundCourseParticipationId)
+
+        expect(result).toEqual(null)
+
+        expect(courseClientBuilder).toHaveBeenCalledWith(token)
+        expect(courseClient.findParticipation).toHaveBeenCalledWith(notFoundCourseParticipationId)
+      })
+    })
+
+    describe('when the course client throws any other error', () => {
+      it('re-throws the error', async () => {
+        const clientError = createError(501)
+        courseClient.findParticipation.mockRejectedValue(clientError)
+
+        const courseParticipationId = faker.string.uuid()
+
+        await expect(() => service.getParticipation(token, courseParticipationId)).rejects.toHaveProperty(
+          'userMessage',
+          `Error fetching course participation with ID ${courseParticipationId}.`,
+        )
+
+        expect(courseClientBuilder).toHaveBeenCalledWith(token)
+        expect(courseClient.findParticipation).toHaveBeenCalledWith(courseParticipationId)
+      })
+    })
```

Comment:
Do we need these two tests? We don't have these error handling tests for other internal API service/client methods, only for the ones that hit external APIs (prisoner offender search and prison register)

Reply:
I think they're useful, why wouldn't we test them? Open to discussion though